src/Core/Users/Actions/CreateUser.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace GetCandy\Api\Core\Users\Actions;
+
+use GetCandy;
+use GetCandy\Api\Core\Customers\Actions\FetchDefaultCustomerGroup;
+use GetCandy\Api\Core\Customers\Models\Customer;
+use GetCandy\Api\Core\Customers\Models\CustomerGroup;
+use GetCandy\Api\Core\Foundation\Actions\DecodeIds;
+use GetCandy\Api\Core\Languages\Actions\FetchDefaultLanguage;
+use GetCandy\Api\Core\Languages\Actions\FetchEnabledLanguageByCode;
+use GetCandy\Api\Core\Users\Resources\UserResource;
+use Lorisleiva\Actions\Action;
+
+class CreateUser extends Action
+{
+    /**
+     * Determine if the user is authorized to make this action.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the action.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'firstname' => 'required|string',
+            'lastname' => 'required|string',
+            'email' => 'required|email',
+            'password' => 'required|string|min:6',
+            'language' => 'nullable|string',
+            'details' => 'nullable|array',
+            'customer_groups' => 'nullable|array',
+        ];
+    }
+
+    /**
+     * Execute the action and return a result.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        $userModel = config('auth.providers.users.model', User::class);
```

Comment:
I saw somewhere we have `GetCandy::getUserModel()` available, should we be using this helper here?